### PR TITLE
added load more instead of next/prev page

### LINKED ISSUE #14: Add a "Load More" button.
Instead of going to the next page, create a "Load More" button, that can update the existing list.

### PR DESCRIPTION
**Pull request by Sigurdurhelga**
Fixes #14 
-

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -10,23 +10,21 @@ class App extends Component{
       urlList: [],
       repoName: [],
       repoUrl: []
-
     };
   }
 
   componentDidMount() {
-    this.callPage(1);
+    this.callPage(1,10);
```
Comment (akramkazmi71) [inline src/App.js]: Can we change this to 30? Since github provides 30 data in an API call.

File: src/App.js [hunk 2]
```diff
@@ -38,57 +36,47 @@ class App extends Component{
...
     const repo=[]
+    let {urlList, repoName, size} = this.state;
     for (const [index, value] of tempArr.entries()){
       urls.push(value)
       let urlSplit=value.replace(/https:\/\/github.com\/|\/issues\/[0-9]+/g,'')
       var repName=urlSplit.split("/")
-      repo.push(<div class="card"><p key={index}><a href={value}>{index+1}. {repName[1]}</a></p></div>)
+      repo.push(<div class="card"><p key={size+index}><a href={value}>{size+index+1}. {repName[1]}</a></p></div>)
     }
+    console.log("got new urls concatting",urls);
```
Comment (akramkazmi71) [inline src/App.js]: Can you remove this? We don't want to keep log statements.

File: src/App.js [hunk 3]
```diff
@@ -35,60 +35,60 @@ class App extends Component{
...
+      urlList: urlList,
+      size: size+totalCount,
+      repoName: repoName
     })
   }
 
   render() {
-    
+
+    let load_button;
```
Comment (arshadkazmi42) [inline src/App.js]: I was thinking, lets move this check in a separate function.

Let's have this in the class
```
hasMore() {
  return this.state.size < this.state.total_count;
}
```

File: src/App.js [hunk 4]
```diff
@@ -35,60 +35,60 @@ class App extends Component{
...
     return(
       <div class="background">
         <a href="https://github.com/akramkazmi71/hacktoberfest-xrepos">
         <img border="0" alt="Github" src="https://github.blog/wp-content/uploads/2008/12/forkme_right_darkblue_121621.png?resize=149%2C149" width="150" height="150"></img></a>
         <h2><u>Hacktoberfest Excluded Repositories</u></h2>
         <div class="container">
          {this.state.repoName}
-          {this.state.pageNumber>1 && <button class="nextPrev" onClick={this.previousPage}>Previous Page</button>}
-          <button class="nextPrev" onClick={this.nextPage}>Next Page</button> 
+         {load_button}
```
Comment (arshadkazmi42) [inline src/App.js]: Here let's change 
```
{load_button}
```
to
```
{this.hasMore() && <button class="nextPrev" onClick={this.loadMore}>Load More</button>;}
```

File: src/App.js [hunk 5]
```diff
@@ -35,60 +35,56 @@ class App extends Component{
...
-      urlList: urls,
-      size: totalCount,
-      repoName: repo
+      urlList: urlList,
+      size: size+totalCount,
+      repoName: repoName
     })
   }
 
+  hasMore(){
```
Comment (arshadkazmi42) [inline src/App.js]: Sorry about this, we will have to use arrow functions here, so as to get access to `this` in the funciton.
Can you update this this to `hasMore = () => {`

**Pull request merged**